src/features/ModelPlan/ModelToOperations/_components/Modal/MilestoneForm/index.tsx

```diff
@@ -0,0 +1,302 @@
+              </FormGroup>
+            )}
+          />
+
+          <Controller
+            name="subcategory"
+            control={control}
+            rules={{
+              required: true,
+              validate: value => value !== 'default'
+            }}
+            render={({ field: { ref, ...field } }) => (
+              <FormGroup className="margin-top-0 margin-bottom-2">
+                <Label
+                  htmlFor={convertCamelCaseToKebabCase(field.name)}
+                  className="mint-body-normal maxw-none margin-bottom-1"
+                >
+                  <Trans
+                    i18nKey={t('modal.milestone.selectSubcategory.label')}
+                    components={{
+                      s: <span className="text-secondary-dark" />
+                    }}
+                  />
+                </Label>
+
+                <Select
+                  {...field}
+                  id={convertCamelCaseToKebabCase(field.name)}
+                  value={field.value || ''}
+                  defaultValue="default"
+                  disabled={watch('primaryCategory') === 'default'}
+                >
+                  {[selectOptions[0], ...mappedSubcategories].map(option => {
+                    return (
+                      <option
+                        key={`sort-${convertCamelCaseToKebabCase(option.label)}`}
+                        value={option.value}
+                      >
+                        {option.label}
+                      </option>
+                    );
+                  })}
+                </Select>
+              </FormGroup>
+            )}
+          />
+
+          <Controller
+            name="name"
+            control={control}
+            rules={{
+              required: true
+            }}
+            render={({ field: { ref, ...field } }) => (
+              <FormGroup className="margin-top-0 margin-bottom-2">
+                <Label
+                  htmlFor={convertCamelCaseToKebabCase(field.name)}
+                  className="mint-body-normal maxw-none margin-bottom-1"
+                >
+                  <Trans
+                    i18nKey={t('modal.milestone.milestoneTitle')}
+                    components={{
+                      s: <span className="text-secondary-dark" />
+                    }}
+                  />
+                </Label>
+
+                <TextInput
+                  type="text"
+                  {...field}
+                  id={convertCamelCaseToKebabCase(field.name)}
+                  value={field.value || ''}
+                />
+              </FormGroup>
+            )}
+          />
+        </Fieldset>
+        <Alert type="info" slim className="margin-bottom-2">
+          <Trans
+            i18nKey={t('modal.milestone.alert.info')}
+            components={{
+              s: <span className="text-underline text-primary-light" />
+              // TODO: Add a link to the documentation
+            }}
```

Comment:
You can add the milestone library link now.  It will be - `/models/${modelID}/collaboration-area/model-to-operations/milestone-library`